x/supplier/types/message_stake_supplier.go

```diff
@@ -12,23 +12,52 @@ const TypeMsgStakeSupplier = "stake_supplier"
 var _ sdk.Msg = (*MsgStakeSupplier)(nil)
 
 func NewMsgStakeSupplier(
-	address string,
+	senderAddress string,
+	ownerAddress string,
+	supplierAddress string,
 	stake sdk.Coin,
 	services []*sharedtypes.SupplierServiceConfig,
 ) *MsgStakeSupplier {
 	return &MsgStakeSupplier{
-		Address:  address,
-		Stake:    &stake,
-		Services: services,
+		Sender:       senderAddress,
+		OwnerAddress: ownerAddress,
+		Address:      supplierAddress,
+		Stake:        &stake,
+		Services:     services,
 	}
 }
 
 func (msg *MsgStakeSupplier) ValidateBasic() error {
+	// Validate the owner address
+	if _, err := sdk.AccAddressFromBech32(msg.OwnerAddress); err != nil {
+		return ErrSupplierInvalidAddress.Wrapf("invalid owner address %s; (%v)", msg.OwnerAddress, err)
+	}
+
+	// Ensure the sender address matches the owner address or the operator address.
+	if msg.Sender != msg.OwnerAddress && msg.Sender != msg.Address {
+		return ErrSupplierInvalidAddress.Wrapf(
+			"sender address %s does not match owner address %s or supplier address %s",
+			msg.Sender,
+			msg.OwnerAddress,
+			msg.Address,
+		)
+	}
+
 	// Validate the address
 	if _, err := sdk.AccAddressFromBech32(msg.Address); err != nil {
```

Comment:
Reoder these checks.

1. Check if owner address is bech32
2. Check if operator is bech32
3. Compare signers against both of them